src/sentry/static/sentry/app/components/teams/createTeamForm.tsx

```diff
@@ -12,12 +12,21 @@ type Payload = {
 
 type Props = {
   organization: Organization;
-  onSubmit?: (data: object, onSuccess: Function, onError: Function) => void;
+  onSubmit?: (data: Payload, onSuccess: Function, onError: Function) => void;
   onSuccess?: (data: Payload) => void;
   formProps?: Partial<typeof Form>;
 };
 
 export default class CreateTeamForm extends React.Component<Props> {
+  handleSubmit = (data: Record<string, any>, onSuccess, onError) => {
+    const {onSubmit} = this.props;
+    if (typeof onSubmit !== 'function') {
+      return;
+    }
+
+    onSubmit(data as Payload, onSuccess, onError);
+  };
```

Comment:
Wanted to avoid adding this, but wasn't sure of a better way to cast `data` as `Payload`